sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/implementation/Utility.java

```diff
@@ -480,24 +499,84 @@ public static ExtractKeyPhrasesResultCollection toExtractKeyPhrasesResultCollect
         return IterableStream.of(analyzeHealthcareEntitiesResults);
     }
 
-//    /**
-//     * Helper function that parse healthcare entity index from the given entity reference string.
-//     * The entity reference format is "#/results/documents/0/entities/3".
-//     *
-//     * @param entityReference the given healthcare entity reference string.
-//     *
-//     * @return the healthcare entity index.
-//     */
-//    private static Integer getHealthcareEntityIndex(String entityReference) {
-//        if (!CoreUtils.isNullOrEmpty(entityReference)) {
-//            int lastIndex = entityReference.lastIndexOf('/');
-//            if (lastIndex != -1) {
-//                return Integer.parseInt(entityReference.substring(lastIndex + 1));
-//            }
-//        }
-//        throw LOGGER.logExceptionAsError(
-//            new RuntimeException("Failed to parse healthcare entity index from: " + entityReference));
-//    }
+    public static HealthcareEntityAssertion toHealthcareEntityAssertion(HealthcareAssertion healthcareAssertion) {
+        final Association association = healthcareAssertion.getAssociation();
+        final Certainty certainty = healthcareAssertion.getCertainty();
+        final Conditionality conditionality = healthcareAssertion.getConditionality();
+
+        final HealthcareEntityAssertion entityAssertion = new HealthcareEntityAssertion();
+        if (association != null) {
+            HealthcareEntityAssertionPropertiesHelper.setAssociation(entityAssertion,
+                EntityAssociation.fromString(association.toString()));
+        }
+        if (certainty != null) {
+            HealthcareEntityAssertionPropertiesHelper.setCertainty(entityAssertion,
+                toCertainty(certainty));
+        }
+        if (conditionality != null) {
+            HealthcareEntityAssertionPropertiesHelper.setConditionality(entityAssertion,
+                toConditionality(conditionality));
+        }
+        return entityAssertion;
+    }
+
+    private static EntityCertainty toCertainty(Certainty certainty) {
```

Comment:
Can these conversions not be done using `fromString` ?

Reply:
EntityCertainty is not an enum class which doesn't have a fromString() method.